actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -834,7 +835,7 @@ def hash_filter(params, filter)
   #         # It's mandatory to specify the nested attributes that should be whitelisted.
   #         # If you use `permit` with just the key that points to the nested attributes hash,
   #         # it will return an empty hash.
-  #         params.require(:person).permit(:name, :age, pets_attributes: [ :name, :category ])
+  #         params.require(:person).permit(:name, :age, pets_attributes: [ :id, :name, :category ])
```

Comment:
I was thinking along these lines: you will probably always want `:id`, because otherwise you'll get new records when updating - the exact thing I was explaining. So if you're not reading carefully and copy-paste it'll just work. `:_destroy` on the other end might not always be wanted and also requires configuration on your model, which is out of scope for ActionController documentation, hence the reference to the proper docs.